### PR TITLE
feat: always respond with 200 but include status key

### PR DESCRIPTION
**Pull request by petarvujovic98**
image

### CODE REVIEW

File: host/src/server/api/v1/mod.rs [hunk 1]
```diff
@@ -58,6 +59,24 @@ pub struct ProofResponse {
...
+        .into_response()
+    }
+}
+
+#[derive(Debug, Serialize, ToSchema)]
+#[serde(tag = "status", rename_all = "lowercase")]
+#[allow(dead_code)]
+pub enum Status {
+    Ok { data: ProofResponse },
+    Error { error: String, message: String },
```
Comment (smtmfft) [inline host/src/server/api/v1/mod.rs:77]: is {error: String} enough @YoGhurt111 ? Looks error & message are dup

**Review by smtmfft — APPROVED** [review]
LGTM, just thought maybe 1 more round of message beautify is needed.

**Comment by smtmfft** [comment]
BTW: I made a new rlpHexBytes serde. so the header will become 0x1234 instead of [18, 52]. will have a PR later.

File: host/src/server/api/v1/mod.rs (hunk 1 above)
Comment (petarvujovic98) [reply host/src/server/api/v1/mod.rs:77]: The `error` key is the type of error (`Io`, `Db`...) where message is details of that particular instance of the error (e.g. why `Io` failed or why `Db` failed)

**Comment by YoGhurt111** [comment]
> BTW: I made a new rlpHexBytes serde. so the header will become 0x1234... instead of [18, 52,...]. will have a PR later. What's your opinion? @YoGhurt111 ?


I don't care about `header`, tbh I only care about `proof` and `message`. So I'm fine with either one.

**Comment by smtmfft** [comment]
Can we merge this one ASAP? then client can integrate & give feedbacks if exist

**Pull request merged**